lib/events.js

```diff
@@ -163,24 +161,21 @@ function emitMany(handler, isFn, self, args) {
   }
 }
 
-EventEmitter.prototype.emit = function emit(type) {
-  var er, handler, len, args, i, events, domain;
-  var needDomainExit = false;
-  var doError = (type === 'error');
+EventEmitter.prototype.emit = function emit(type, ...args) {
+  let doError = (type === 'error');
 
-  events = this._events;
-  if (events)
-    doError = (doError && events.error == null);
+  const events = this._events;
+  if (events !== undefined)
```

Comment:
Changes like this really warrant a CITGM run.

Reply:
Yeah, I completely agree. 👍  I tried to flag it in my description of the PR but it's a bit of a wall of text.